rust/src/lib.rs

```diff
@@ -852,6 +852,36 @@ pub extern "C" fn zcashlc_get_received_memo_as_utf8(
     unwrap_exc_or_null(res)
 }
 
+/// Returns the memo for a received note, if it is known and a valid UTF-8 string.
+///
+/// The note is identified by its row index in the `received_notes` table within the data
+/// database.
+///
+/// Call `zcashlc_byte_array_free` on the returned pointer when you are finished with it.
```

Comment:
The documentation doesn't say what the returned memo length is, or what should be passed into `zcashlc_byte_array_free`. I'd prefer that the caller provide `memo_bytes_ret: *mut u8` as an argument, and that the bytes be written into that. This would remove the need for `zcashlc_byte_array_free`.